feat(select): component can be used outside ion-item

liamdebeasi opened the pull request:
<!-- Please refer to our contributing documentation for any questions on submitting a pull request, or let us know here if you need any help: https://ionicframework.com/docs/building/contributing -->

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures


## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. --> 

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. --> 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying. -->


<!-- Issues are required for both bug fixes and features. -->
Issue URL: Internal ticket


## What is the new behavior?
<!-- Please describe the behavior or changes that are being added by this PR. -->

- `ion-select` can now be used outside of `ion-item` with a new property to specify the label.
This PR is a combination of the following:

https://github.com/ionic-team/ionic-framework/pull/26470
https://github.com/ionic-team/ionic-framework/pull/26502
https://github.com/ionic-team/ionic-framework/pull/26504
https://github.com/ionic-team/ionic-framework/pull/26505
https://github.com/ionic-team/ionic-framework/pull/26506
https://github.com/ionic-team/ionic-framework/pull/26570


## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

sean-perkins commented on core/src/components/select/test/fill/index.html:
```diff
@@ -0,0 +1,151 @@
+
+          <div class="grid-item">
+            <h2>Solid, Floating Label</h2>
+            <ion-select fill="solid" label="Fruit" label-placement="floating" value="apple"></ion-select>
```
~ Is there a reason that some of the examples have options and others do not? I know the test does not use this template, but it might be helpful if all the examples render at least 1 option for a consistent display (unless we are wanting to demonstrate a state of no options).

Adding options would also allow manual verification of the visual state after selecting a value.

liamdebeasi replied:
I think that was an oversight on my part. I pushed a commit that ensures all selects have values.

sean-perkins reviewed (commented):
Should we have examples/tests for the different interface appearances of `ion-select` and the modern rendering? 

The legacy tests show examples with `interface="popover"`.

liamdebeasi commented:
> Should we have examples/tests for the different interface appearances of ion-select and the modern rendering?
> The legacy tests show examples with interface="popover".

I added tests. Will re-run screenshot jobs shortly.

amandaejohnston commented:
I think it should still appear on the end side, since that's where the actual select appears to be. Otherwise it can easily look like the popover is just opening under empty space, which is strange. IMO behavior should match with how the legacy select acts when placed within an item, since that's what it looks like, even when the new syntax isn't used with an item as well.

liamdebeasi commented:
Ok I fixed a handful of things:

1. Noticed an issue where justification wasn't working. This broke as a result of some code I missed when resolving a merge conflict.
2. Fixed an issue Amanda noted where the popover is not aligned correctly. Should now be aligned relative to the placeholder/icon.
3. Fixed an issue where the popover does not take up the full width when using outline/solid styles
4. Fixed an issue where the chevron was rotating on open on iOS. This was due to some code that slipped through when fixing a merge conflict.

Updating screenshots now and will re-request another review when that is done.
